pylint/testutils/configuration_test.py

```diff
@@ -0,0 +1,101 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+
+"""Utility function for configuration testing."""
+import copy
+import json
+import logging
+import unittest
+from pathlib import Path
+from typing import Any, Dict, Tuple, Union
+from unittest.mock import Mock
+
+from pylint.lint import Run
+
+USER_SPECIFIC = str(Path(__file__).parent.parent.parent)
+
+
+def get_expected_or_default(pyproject_toml_path: str, suffix: str, default: Any) -> str:
+    def get_path_according_to_suffix() -> Path:
+        path = Path(pyproject_toml_path)
+        return path.parent / f"{path.stem}.{suffix}"
+
+    expected = default
+    expected_result_path = get_path_according_to_suffix()
+    if expected_result_path.exists():
+        with open(expected_result_path, encoding="utf8") as f:
+            expected = f.read()
+        logging.info("%s exists.", expected_result_path)
```

Comment:
No I don't think so. Here it's very helpful to see what's happening (especially to realize your file is not taken into account after a misspell of the file name). We can't print because we have a lot of writing into streams and the output of the program is checked during the test so printing messes with the result. I'll add a comment.

Reply:
Ah, makes sense. Comment would be helpful indeed 😄

Reply:
da4c8feb419a54544f076258bf4116bd8cd21fd3